mixer/adapter/signalfx/tracing.go

```diff
@@ -28,6 +28,7 @@ import (
 	octrace "go.opencensus.io/trace"
 
 	"istio.io/istio/mixer/pkg/adapter"
+	"istio.io/istio/mixer/pkg/attribute"
```

Comment:
Adapters aren't supposed to deal with attributes, and as such I think importing this package here indicates a problem.

Reply:
FYI It is used to import `attribute.StringMap`. 

Here is the problem. Trace span uses a field with a dynamic value in the map values:
```
https://github.com/istio/istio/blob/master/mixer/template/tracespan/tracespan.proto#L109
```

Existing code generation works by places the raw attribute value in the map. That means the string map shows up as a value, and `net.IP` is turned into raw bytes. This is nothing new, the code simply didn't work before since it was casting to a wrong type in both cases.

Reply:
What is the wire format used for OOP adapters for IP and email addresses?

Reply:
For OOP, the value holds the protobuf dynamic value: https://github.com/istio/istio/blob/master/mixer/template/tracespan/tracespan_handler_service.proto#L154

This code is for in-process adapter, which doesn't look like it would work out of the box.